geomesa-tools/src/main/scala/org/locationtech/geomesa/tools/data/ManagePartitionsCommand.scala

```diff
@@ -110,22 +111,25 @@ object ManagePartitionsCommand {
         val version = s"v${index.version}"
         (index, name, attributes, version)
       }
-      val indicesAndTables = tables.map { table =>
-        val matchedByName = indexNames.filter { case (_, n, _, v) => table.contains(n) && table.contains(v) }
-        if (matchedByName.lengthCompare(1) == 0) {
-          (matchedByName.head._1, table)
-        } else {
-          val matchedByAttributes = matchedByName.filter(m => table.contains(m._3))
-          if (matchedByAttributes.lengthCompare(1) == 0) {
-            (matchedByAttributes.head._1, table)
+      val tableToIndexName = indexNames
+        .toList
+        .sortWith(_._3.size > _._3.size) // Sorted so we get best match first
+        .foldLeft(Map[String,(GeoMesaFeatureIndex[_,_],String,String,String)]()){
```

Comment:
nit: this works, but probably would be simpler code to build up a mutable map in a foreach loop. don't need to change it though